pkg/asset/installconfig/kubevirt/validation.go

```diff
@@ -17,6 +18,7 @@ func Validate(ic *types.InstallConfig, client Client) error {
 	allErrs = append(allErrs, validateNamespace(kubevirtPlatform.Namespace, client, fldPath.Child("namespace"))...)
 	allErrs = append(allErrs, validateStorageClassExistsInInfraCluster(kubevirtPlatform.StorageClass, client, fldPath.Child("storageClass"))...)
 	allErrs = append(allErrs, validateNetworkAttachmentDefinitionExistsInNamespace(kubevirtPlatform.NetworkName, kubevirtPlatform.Namespace, client, fldPath.Child("networkName"))...)
+	allErrs = append(allErrs, validateHcoFeatureGates(client, fldPath.Child("featureGates"))...)
```

Comment:
The field path extension should be done using valid field name from the platform fields:
https://github.com/openshift/installer/blob/8fdf359d8edbc0ad52a0fe3adaf801c56f28327a/pkg/types/kubevirt/platform.go#L4
Need to consider which field is the appropriate one to perform this validation for

Reply:
I concur with @nirarg. You cannot use a field path of `featureGates` here because the install config does not have that field. The reason that there is a question about what the field name should be is because this is not the right place to be validating whether the feature gate is enabled. This `Validate` function is intended to validate the install config in ways that require access to the platform on which the install is being performed. Checking for whether feature gates are enabled or not is not a validation on the install config. I suggest that the feature-gate validation be moved to the [Platform Provision Check asset](https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go).